### PR TITLE
Gorger Feast is AOE + lower cooldown

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Atomization: https://github.com/tgstation/TerraGov-Marine-Corps/pull/13317
Feast is now AOE + lower cooldown (30 seconds from 180)
## Why It's Good For The Game
Copypaste: "Gorger is really uncomfortable to play in general, these are merely QoL considering what state he is in right now, Gorger cannot sustain effectively neither offensive nor support abilities, some abilities are very rarely used in general, this tries to correct that (very successfully, considering RU experience with it). Probably needs a testmerge to estimate how good/bad these changes are. These changes are present on RU for 3 months without any balance issues."
## Changelog
:cl:
balance: Gorger Feast is AOE
balance: Gorger Feast now 30 seconds (180 seconds before)
/:cl:

### CODE REVIEW

File: code/datums/status_effects/xeno_buffs.dm [hunk 1]
```diff
@@ -616,15 +616,20 @@
...
-	adjustOverheal(X, heal_amount / 2)
-	X.use_plasma(plasma_drain)
+	var/heal_amount = xeno_caster.maxHealth*0.08
+	for(var/mob/living/carbon/xenomorph/affected_xeno AS in GLOB.hive_datums[xeno_caster.hivenumber].xenos_by_zlevel["[X.z]"])
+		if(affected_xeno.faction != xeno_caster.faction)
+			continue
+		if(!line_of_sight(xeno_caster, affected_xeno, 4))
+			continue
+		HEAL_XENO_DAMAGE(xeno_caster, heal_amount, FALSE)
+		adjustOverheal(xeno_caster, heal_amount / 2)
```
Comment (Waselon) [inline code/datums/status_effects/xeno_buffs.dm:631]: should be affected_xeno instead

**Comment by Pariah919** [comment]
New xander arc?
I've honestly never seen Gorger as an issue.

**Comment by Cerrik** [comment]
I think this would be a cool buff, but having both 30 second cooldown and AoE might be too much together.  I'm not confident I'm right though, because Feast obviously uses up so much blood that even when the cooldown ends, you probably don't have enough to do a meaningful Feast.

I'm mostly thinking how the Gorger primo is so similarly selfish to Feast that I think changing one of them is a cool idea.

**Comment by yyzsong** [comment]
AOE healing is *probably* too strong for a baseline gorger ability (I'd like it to be the primo, if anything) and a change in cooldown is meaningless because you won't ever be able to get back up to feast levels of blood in 60 seconds (unless the other prs get merged (they wont))

File: code/datums/status_effects/xeno_buffs.dm [hunk 2]
```diff
@@ -616,15 +616,20 @@
...
-	var/heal_amount = X.maxHealth*0.08
-	HEAL_XENO_DAMAGE(X, heal_amount, FALSE)
-	adjustOverheal(X, heal_amount / 2)
-	X.use_plasma(plasma_drain)
+	var/heal_amount = xeno_caster.maxHealth*0.08
+	for(var/mob/living/carbon/xenomorph/affected_xeno AS in GLOB.hive_datums[xeno_caster.hivenumber].xenos_by_zlevel["[xeno_caster.z]"])
+		if(affected_xeno.faction != xeno_caster.faction)
+			continue
+		if(!line_of_sight(xeno_caster, affected_xeno, 4))
+			continue
```
Comment (psykzz) [inline code/datums/status_effects/xeno_buffs.dm:629]: I dont know if line of sight has been improved at all, but might be worth doing a range check before this to short circuit the los check

Comment (Lumipharon) [reply code/datums/status_effects/xeno_buffs.dm:629]: yeah if you're running los to every xeno, its probs more performant to check dist first lol

**Pull request closed without merging**